cumulusci/utils/http/multi_request.py

```diff
@@ -21,19 +24,41 @@ def __enter__(self, *args):
     def __exit__(self, *args):
         self.session.close()
 
-    def _async_request(self, path, method, json=None, headers=None):
-        headers = {**(headers or {}), "Accept-Encoding": "gzip"}
+    def _async_request(
+        self, url: str, method: str, json: object = None, httpHeaders: dict = None
+    ):
+        "Make an async HTTP request and return a future"
+        headers = {**(httpHeaders or {}), "Accept-Encoding": "gzip"}
         return self.session.request(
             method=method,
-            url=self.base_url + path.lstrip("/"),
+            url=self.base_url + url.lstrip("/"),
             headers=headers,
             json=json,
+            timeout=30,
+        )
+
+    def do_requests(self, requests: T.Iterable[T.Dict]):
+        def catch(future, futures_to_requests):
```

Comment:
You could move this down below line 49 and let futures_to_requests be in the closure

Reply:
I decided that maybe its pushing the boundary of the size of a a good closure and had very few dependencies on local state, so I just turned it into a normal function. 08b7f7d6e